### PR TITLE
Test for basic endpoints

### PR DESCRIPTION
**Pull request by MiSobecki**

### CODE REVIEW

File: src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@AutoConfigureTestDatabase
+@AutoConfigureDataJpa
+class OfferControllerSpock extends Specification {
+
+    private final String offerBody = """
```
Comment (DaDudek) [inline src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy:29]: it seems like this is some const so should be static final and named OFFER_BODY

File: src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    private MockMvc mockMvc
+
+    @Autowired
+    private OfferRepository offerRepository
+
+    private Offer offer
+
+    void setup() {
+        offer = new Offer(
+                BigDecimal.valueOf(30).setScale(2),
```
Comment (DaDudek) [inline src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy:52]: why we have setScale(2) here and setScale(0) on mr with alter table?

File: src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        when: "we perform post request with offer given in body"
+        def result = mockMvc
+                .perform(post("/api/offers")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(offerBody))
+
+        then: "response status is equal to Created and we can find created offer in db"
+        result.andExpect(status().isCreated())
+
+        assertThat(offerRepository.findAll())
```
Comment (DaDudek) [inline src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy:71]: why you use assertThat?

File: src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                .extracting(
+                        Offer.&getPrice as Function,
+                        Offer.&getDate as Function,
+                        Offer.&getDescription as Function,
+                        Offer.&getAvailability as Function)
+                .contains(tuple(
+                        offer.getPrice(),
+                        offer.getDate(),
+                        offer.getDescription(),
+                        offer.getAvailability()))
```
Comment (DaDudek) [inline src/test/groovy/com/example/foode/Offer/OfferControllerSpock.groovy:81]: this block looks really weird. Why its do it like that?

File: src/main/java/com/example/foode/Offer/Offer.java [hunk 5]
```diff
@@ -35,14 +36,45 @@ public class Offer {
     @Column(length = 200, nullable = false)
     private String availability;
 
-    @ManyToOne()
+    @JsonBackReference
```
[the remark this answers is not in the document]
Comment (MiSobecki) [reply src/main/java/com/example/foode/Offer/Offer.java:39]: But there is @ManyToOne(fetch = FetchType.LAZY) instead.

**Pull request merged**